### PR TITLE
bitcoincoid: less strict error checking for public API

### PR DESCRIPTION
**Pull request by mkutny**
Properly fixes #1707 .
Actually, I checked that PR against private API only... and Travis skipped checks for public API too.

### CODE REVIEW

**Comment by mkutny** [comment]
And should fix #1678. Still don't have open orders there, maybe when will see them it will need further fixes.

**Comment by mkutny** [comment]
Quite many commits, I know... Just don't have access to bitcoincoid from my working machine, had to push/pull to my data center at Jakarta to test.

**Comment by mkutny** [comment]
Transpiler in my head is broken. Linter in no help too. Still have few minutes for fix today if you tell me what's wrong with the syntax.

File: js/bitcoincoid.js [hunk 1]
```diff
@@ -256,16 +255,16 @@ module.exports = class bitcoincoid extends Exchange {
...
     async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
-        if (!symbol)
-            throw new ExchangeError (this.id + ' fetchOpenOrders requires a symbol');
         await this.loadMarkets ();
-        let market = this.market (symbol);
-        let request = {
-            'pair': market['id'],
-        };
+        let market = symbol ? this.market (symbol) : undefined;
+        let request = market ? { 'pair': market['id'] } : {};
```
Comment (kroitor) [inline js/bitcoincoid.js]: it won't like these two lines:

```
let market = undefined;
if (symbol)
    market = this.market (symbol);
```

```
let request = {};
if (market)
    request['pair'] = market['id'];
```

File: js/bitcoincoid.js [hunk 2]
```diff
@@ -256,16 +255,16 @@ module.exports = class bitcoincoid extends Exchange {
...
         let response = await this.privatePostOpenOrders (this.extend (request, params));
-        let orders = this.parseOrders (response['return']['orders'], market, since, limit);
-        return this.filterOrdersBySymbol (orders, symbol);
+        // { success: 1, return: { orders: null }}
+        let raw = response['return']['orders'];
+        if (!raw)
+            return [];
+        let orders = this.parseOrders (raw, market, since, limit);
+        return market ? this.filterOrdersBySymbol (orders, symbol) : orders;
```
Comment (kroitor) [inline js/bitcoincoid.js]: And this one, basically it's overly sensitive to ternary conditionals, can only recognize the most simple ones...
```(market) ? noBracketsA : noBracketsB```

File: js/bitcoincoid.js (hunk 1 above)
Comment (mkutny) [reply js/bitcoincoid.js]: How do these differ from this line?
https://github.com/ccxt/ccxt/blob/f56b5c2f94a52f4bf5da013c5c545273856ab2be/js/liqui.js#L474

File: js/bitcoincoid.js (hunk 2 above)
Comment (mkutny) [reply js/bitcoincoid.js]: This one I removed. `filtersOrdersBySymol` is robust against `undefined` symbol.

File: js/bitcoincoid.js (hunk 1 above)
Comment (kroitor) [reply js/bitcoincoid.js]: this.market() has round brackets, this breaks the transpiler, as round brackets add ambiguity (very hard to determine when one part of the expression starts and the other ends with pure regexes like that).

**Comment by kroitor** [comment]
No worries, I'll merge it. Thx!

**Comment by mkutny** [comment]
Just checked the linter.
We can enforce multiline ternary if it helps transpiling:
```javascript
var foo = bar > baz ?
    value1 :
    value2;
```

**Comment by kroitor** [comment]
> if it helps transpiling:

In fact, it's even worse, as multiline means that we have to rework the padding as well, which is harder, than doing it line-to-line.... But I think it's better to make the transpiler work with those expressions, will tweak the regexes to tolerate the brackets. Almost merged this PR already )

**Comment by mkutny** [comment]
Ok, one more tweak.

**Pull request merged**